pkg/tfplan/proposed_new.go

```diff
@@ -0,0 +1,232 @@
+// Copyright 2016-2023, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tfplan
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-go/tfprotov6"
+	"github.com/hashicorp/terraform-plugin-go/tftypes"
+	"github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfplan/internal/tf/configs/configschema"
+	"github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfplan/internal/tf/plans/objchange"
+	"github.com/zclconf/go-cty/cty"
+	ctypack "github.com/zclconf/go-cty/cty/msgpack"
+)
+
+func ProposedNew(
```

Comment:
This is the public entry point for this package. Can we add a comment explaining why it exists and what it does.

If we don't believe another repository could consume this function, we should mark it as unstable for consumption. We should be clear on when we adopt a backwards compatibility requirement.